### PR TITLE
DFP video ad server implementation

### PR DESCRIPTION
**Pull request by spra85**

### CODE REVIEW

**Comment by spra85** [comment]
[Modify the vastUrl endpoint to instead point at DFP video for targeting](https://trello.com/c/zqWJqRRu/6-modify-the-vasturl-endpoint-to-instead-point-at-dfp-video-for-targeting)

File: elements/bulbs-video/components/revealed.js [hunk 1]
```diff
@@ -172,29 +173,47 @@ export default class Revealed extends React.Component {
...
+    baseUrl += '&unviewed_position_start=1';
+    baseUrl += `&url=${window.document.referrer}`;
+    baseUrl += '&description_url=';
+    baseUrl += `&correlator=${new Date().getTime()}`;
+
+    // dfp_pagetype=home&dfp_site=onion&kuid=qxw4bky4u
+
+    let customParamValues = '';
+    customParamValues += `video_site=${videoMeta.channel_slug}`;
+    customParamValues += `&dfp_specialcoverage=${videoMeta.targeting.dfp_specialcoverage}`
```
Comment (spra85) [inline elements/bulbs-video/components/revealed.js]: Wouldn't you want to also wrap this behind the conditional below?

File: elements/bulbs-video/components/revealed.js [hunk 2]
```diff
@@ -172,29 +173,47 @@ export default class Revealed extends React.Component {
...
+    customParamValues += `&dfp_campaign_id=${videoMeta.targeting.dfp_campaign_id}`
+    customParamValues += `&video_id=${videoMeta.targeting.target_video_id}`
```
Comment (spra85) [inline elements/bulbs-video/components/revealed.js]: Most of this should already exist directly on the `videoMeta` object... I believe, which would remove the requirement for the `targeting || {}` from above

**Comment by benghaziboy** [comment]
@spra85 Sorry, got sidetracked by some avc stuff. I think this fulfills the params listed in the card. Need some info on what is going on with `forcedAdZone` and `testVastId` todos.  They are breaking the linter, but I need more detail on them if they are part of this card.

**Comment by spra85** [comment]
@benghaziboy, went ahead and just knocked out that part of the implementation here (98996ae4c56340564f228342fd7dbedddce0399a).  We can set up test links and start to coordinate testing with ops

**Comment by benghaziboy** [comment]
nice. Good for release then?

**Comment by benghaziboy** [comment]
http://dfp-video-testing.test.theonion.com

Need to coordinate with Ad-Ops for testing

**Comment by benghaziboy** [comment]
@theonion/front-end

**Comment by jmelvnsn** [comment]
lgtm

**Comment by spra85** [comment]
@benghaziboy, let's discuss in AM.  Your commit with the dimension should go out in a separate PR, and get released independently of being bundled to this.

**Comment by benghaziboy** [comment]
https://github.com/theonion/bulbs-elements/pull/229

re-opened here after reverting merge.

**Pull request merged**